amxmodx/amxmodx.cpp

```diff
@@ -2558,15 +2558,15 @@ static cell AMX_NATIVE_CALL read_argv_float(AMX *amx, cell *params) /* 1 param *
 		return 0;
 	}
 
-	const char *value = g_fakecmd.notify ? (argc >= 1 && argc < 3 ? g_fakecmd.argv[argc] : "") : CMD_ARGV(argc);
+	const char *value = g_fakecmd.notify ? ((argc >= 1 && argc < 3 && g_fakecmd.argv[argc] != nullptr) ? g_fakecmd.argv[argc]  : "") : CMD_ARGV(argc);
 	float flValue = atof(value);
 
 	return amx_ftoc(flValue);
 }
 
 static cell AMX_NATIVE_CALL read_args(AMX *amx, cell *params) /* 2 param */
 {
-	const char* sValue = g_fakecmd.notify ? (g_fakecmd.argc > 1 ? g_fakecmd.args : g_fakecmd.argv[0]) : CMD_ARGS();
```

Comment:
Unless I'm wrong, this is the correct behavior. `read_args` should include the command + arguments.
In case of using `amxclient_cmd` with empty arguments, it should return at least the command.

Reply:
`read_args` should include only arguments. Remember the popular pattern for handling `say`:
https://github.com/alliedmodders/amxmodx/blob/8f88dc5ac8df471a39bc6904edff4e44a8d46564/plugins/adminchat.sma#L92-L94

Reply:
I checked/tested and it seems you're right. I was pretty sure if no arguments, command would be returned. I vaguely remembered seeing that in ReHLDS, guess not.